respect-lib-shared/src/commonMain/kotlin/world/respect/shared/viewmodel/report/edit/ReportEditViewModel.kt

```diff
@@ -0,0 +1,413 @@
+                                        title = "", series = listOf(
+                                            ReportSeries(
+                                                reportSeriesUid = 1,
+                                                reportSeriesVisualType = ReportSeriesVisualType.BAR_CHART,
+                                                reportSeriesSubGroup = null,
+                                                reportSeriesYAxis = DefaultIndicators.list.first(),
+                                                reportSeriesFilters = emptyList()
+                                            )
+                                        )
+                                    )
+                                    _uiState.update { prev ->
+                                        prev.copy(
+                                            reportOptions = mockedReport
+                                        )
+                                    }
+                                } else {
+                                    _uiState.update { currentState ->
+                                        currentState.copy(
+                                            reportOptions = parsedOptions,
+                                        )
+                                    }
+                                }
+                            }
+
+                            else -> {
+                                val mockedReport = ReportOptions(
+                                    series = listOf(
+                                        ReportSeries(
+                                            reportSeriesUid = 1,
+                                            reportSeriesVisualType = ReportSeriesVisualType.BAR_CHART,
+                                            reportSeriesSubGroup = null,
+                                            reportSeriesYAxis = DefaultIndicators.list.first(),
+                                            reportSeriesFilters = emptyList()
+                                        )
+                                    )
+                                )
+                                _uiState.update { prev ->
+                                    prev.copy(
+                                        reportOptions = mockedReport
+                                    )
+                                }
+                            }
+                        }
+                    }
+                }
+            } catch (e: Exception) {
+                println("Exception $e")
+            }
+
+            viewModelScope.launch {
+                navResultReturner.filteredResultFlowForKey(REPORT_FILTER_RESULT).collect { result ->
+                    val filter = result.result as? ReportFilter
+                    filter?.let {
+                        onFilterChanged(filter, filter.reportFilterSeriesUid.toInt())
+                    }
+                }
+            }
+        }
+    }
+
+    fun manageIndicator() {
```

Comment:
This needs renamed to follow pattern - it's an event e.g. should be onClickManagerIndicators